### PR TITLE
feat(license): add HTTP API for license

### PR DESCRIPTION
**Pull request by thalesmg**

### CODE REVIEW

File: lib-ee/emqx_license/src/emqx_license_http_api.erl [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    {200, License}.
+
+'/license/upload'(post, #{body := #{<<"file">> := Filepath}}) ->
+    case emqx_license:update_file(Filepath) of
+        {error, enoent} ->
+            ?SLOG(error, #{
+                msg => "license_file_not_found",
+                path => Filepath
+            }),
+            {404, <<"file not found">>};
```
Comment (DDDHuang) [inline lib-ee/emqx_license/src/emqx_license_http_api.erl:114]: Response body should be a json.
And error response should be like 
```json
{
    "code": "NOT_FOUND",
    "message": "Bad file XX.lix or something"
}
```

File: lib-ee/emqx_license/src/emqx_license_http_api.erl [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    case emqx_license:update_key(Key) of
+        {error, Error} ->
+            ?SLOG(error, #{
+                msg => "bad_license_key",
+                reason => Error
+            }),
+            {400, <<"bad request">>};
+        {ok, _} ->
+            ?SLOG(info, #{msg => "updated_license_key"}),
+            {200, <<"ok">>}
```
Comment (DDDHuang) [inline lib-ee/emqx_license/src/emqx_license_http_api.erl:139]:
```erlang
{200, #{<<"key">> => Key}}
```
Maybe response the created resource is better.

File: lib-ee/emqx_license/i18n/emqx_license_http_api.conf [hunk 3]
```diff
@@ -0,0 +1,23 @@
+emqx_license_http_api {
+  desc_license_info_api {
+    desc {
+      en: "Get license info"
+      zh: "获取许可证信息"
```
Comment (thalesmg) [inline lib-ee/emqx_license/i18n/emqx_license_http_api.conf:5]: Warning: machine generated.

File: lib-ee/emqx_license/src/emqx_license_http_api.erl (hunk 1 above)
Comment (thalesmg) [reply lib-ee/emqx_license/src/emqx_license_http_api.erl]: Thanks!  I've changed the error returns here: https://github.com/emqx/emqx/pull/8610/commits/50dc90596aa9f22def5d41b7376440320f6d69f7

File: lib-ee/emqx_license/src/emqx_license_http_api.erl (hunk 2 above)
Comment (thalesmg) [reply lib-ee/emqx_license/src/emqx_license_http_api.erl]: I've changed it to return the license here: https://github.com/emqx/emqx/pull/8610/commits/50dc90596aa9f22def5d41b7376440320f6d69f7

File: lib-ee/emqx_license/i18n/emqx_license_http_api.conf [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      en: "License info"
+      zh: "许可证信息"
+    }
+  }
+
+  desc_license_upload_api {
+    desc {
+      en: "Upload a license file or key"
+      zh: "上传许可证文件或钥匙"
+    }
```
Comment (DDDHuang) [inline lib-ee/emqx_license/i18n/emqx_license_http_api.conf:17]: 上传许可证文件或密钥

**Review by DDDHuang — DISMISSED** [review]

File: lib-ee/emqx_license/i18n/emqx_license_http_api.conf (hunk 3 above)
Comment (zhongwencool) [reply lib-ee/emqx_license/i18n/emqx_license_http_api.conf:5]: COOL

**Review by ieQu1 — APPROVED** [review]

File: lib-ee/emqx_license/test/emqx_license_test_lib.erl [hunk 5]
```diff
@@ -47,6 +47,32 @@ test_key(Filename, Format) ->
...
+        customer_type => "10",
+        name => "Foo",
+        email => "contact@foo.com",
+        deployment => "bar-deployment",
+        start_date => "20220111",
+        days => "100000",
+        max_connections => "10"
+    },
+    Values1 = maps:merge(Defaults, Values0),
+    Keys = [
```
Comment (savonarola) [inline lib-ee/emqx_license/test/emqx_license_test_lib.erl:63]: nit: `maps:keys(Defaults)`?

Comment (thalesmg) [reply lib-ee/emqx_license/test/emqx_license_test_lib.erl:63]: In this case, I used a list to make the key order deterministic to match the previous `make_license/1` behavior.

File: lib-ee/emqx_license/src/emqx_license_http_api.erl [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    }.
+
+error_msg(Code, Msg) ->
+    #{code => Code, message => emqx_misc:readable_error_msg(Msg)}.
+
+'/license'(get, _Params) ->
+    License = maps:from_list(emqx_license_checker:dump()),
+    {200, License}.
+
+'/license/upload'(post, #{body := #{<<"file">> := Filepath}}) ->
```
Comment (savonarola) [inline lib-ee/emqx_license/src/emqx_license_http_api.erl:120]: Seems that this should be a rare case, when a user 
* first, puts a node local file with license
* second, send uplodad command through http to the node local HTTP port

Maybe we should handle the license file as an upload, like we did in https://github.com/emqx/emqx/blob/master/apps/emqx_authn/src/emqx_authn_user_import_api.erl

Comment (thalesmg) [reply lib-ee/emqx_license/src/emqx_license_http_api.erl:120]: I think one possibility is that the node is when the node is first booted up, and the license file is only configured in `etc/emqx_enterprise.conf`.  Later updates might come from the HTTP.

Thanks for the reference!  I'll have a look and then improve this with a following PR.  :smiley_cat:

**Pull request merged**